File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        curSum = 0
        while left < right:
            curSum = numbers[left] + numbers[right]
            if curSum == target:
                return [left + 1, right + 1]
            
            elif curSum < target:
                left += 1
            elif curSum > target:
                right -= 1
        

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        List<Integer> res = new ArrayList<>();
        if (root!= null){
            queue.add(root);
        }
        while(!queue.isEmpty()){
            TreeNode cur = new TreeNode(); 
            int levelLength = queue.size();
            for (int i = 0; i< levelLength; i++){
                cur = queue.remove();
                if (cur.left != null ){
                    queue.add(cur.left);
                }
                if (cur.right != null){
                    queue.add(cur.right);
                }
            }
            res.add(cur.val);
        }
        
        return res;
    }
}
